soh/soh/resource/type/Skeleton.cpp

```diff
@@ -65,16 +70,58 @@ void SkeletonPatcher::ClearSkeletons()
+    bool isAlt = CVarGetInteger("gAltAssets", 0);
     for (auto skel : skeletons) 
     {
         Skeleton* newSkel =
             (Skeleton*)LUS::Context::GetInstance()->GetResourceManager()
-                ->LoadResource((isHD ? LUS::IResource::gAltAssetPrefix : "") + skel.vanillaSkeletonPath, true)
+                ->LoadResource((isAlt ? LUS::IResource::gAltAssetPrefix : "") + skel.vanillaSkeletonPath, true)
                 .get();
 
         if (newSkel != nullptr)
             skel.skelAnime->skeleton = newSkel->skeletonData.skeletonHeader.segment;
     }
 }
+
+void SkeletonPatcher::UpdateTunicSkeletons() {
+    bool isAlt = CVarGetInteger("gAltAssets", 0);
+    const char* altPrefix = isAlt ? LUS::IResource::gAltAssetPrefix.c_str() : "";
+    for (auto skel : skeletons) {
+        if (skel.actor != nullptr) // is there actually an actor to check?
+        {
+            if (skel.actor->id == 0) // is this Link?
+            {
+                Skeleton* newSkel;
+                if (LINK_IS_ADULT && CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1 == PLAYER_TUNIC_KOKIRI &&
+                    ResourceGetIsCustomByName(gLinkKokiriTunicSkel)) {
+                    newSkel = (Skeleton*)LUS::Context::GetInstance()
+                                  ->GetResourceManager()
+                                  ->LoadResource((isAlt ? LUS::IResource::gAltAssetPrefix : "") + gLinkKokiriTunicSkel, true)
+                                  .get();
+                } else if (LINK_IS_ADULT && CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1 == PLAYER_TUNIC_GORON &&
+                           ResourceGetIsCustomByName(gLinkGoronTunicSkel)) {
+                    newSkel = (Skeleton*)LUS::Context::GetInstance()
+                                  ->GetResourceManager()
+                                  ->LoadResource((isAlt ? LUS::IResource::gAltAssetPrefix : "") + gLinkGoronTunicSkel, true)
+                                  .get();
+                } else if (LINK_IS_ADULT && CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1 == PLAYER_TUNIC_ZORA &&
+                           ResourceGetIsCustomByName(gLinkZoraTunicSkel)) {
+                    newSkel = (Skeleton*)LUS::Context::GetInstance()
+                                  ->GetResourceManager()
+                                  ->LoadResource((isAlt ? LUS::IResource::gAltAssetPrefix : "") + gLinkZoraTunicSkel, true)
+                                  .get();
+                } else { // child link, no model available
+                    newSkel = (Skeleton*)LUS::Context::GetInstance()
+                                  ->GetResourceManager()
+                                  ->LoadResource(
+                                      (isAlt ? LUS::IResource::gAltAssetPrefix : "") + skel.vanillaSkeletonPath, true)
+                                  .get();
+                }
+
+                if (newSkel != nullptr)
+                    skel.skelAnime->skeleton = newSkel->skeletonData.skeletonHeader.segment;
+            }
+        }
```

Comment:
```suggestion
        if (strcmp(skel.vanillaSkeletonPath.c_str(), &gLinkAdultSkel[OTR_OFFSET]) == 0) {
            Skeleton* newSkel = nullptr;
            if (CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1 == PLAYER_TUNIC_KOKIRI) {
                newSkel = (Skeleton*)LUS::Context::GetInstance()
                                    ->GetResourceManager()
                                    ->LoadResource((isAlt ? LUS::IResource::gAltAssetPrefix : "") + gLinkKokiriTunicSkel, true)
                                    .get();
            } else if (CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1 == PLAYER_TUNIC_GORON) {
                newSkel = (Skeleton*)LUS::Context::GetInstance()
                                    ->GetResourceManager()
                                    ->LoadResource((isAlt ? LUS::IResource::gAltAssetPrefix : "") + gLinkGoronTunicSkel, true)
                                    .get();
            } else if (CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1 == PLAYER_TUNIC_ZORA) {
                newSkel = (Skeleton*)LUS::Context::GetInstance()
                                    ->GetResourceManager()
                                    ->LoadResource((isAlt ? LUS::IResource::gAltAssetPrefix : "") + gLinkZoraTunicSkel, true)
                                    .get();
            }
            if (newSkel != nullptr) {
                skel.skelAnime->skeleton = newSkel->skeletonData.skeletonHeader.segment;
            }
        }
```

This should be a way to remove the need for passing the actor around (and no adult link checks needed either!) the OTR_OFFSET will need to be defined in this case as 7 (the length of the string '__OTR__')

Reply:
Going to look into the alt toggling issue now

Reply:
Also let me know if you want me to PR the changes to your branch or help in any other way with these changes!